src/typedefs.jl

```diff
@@ -73,7 +75,7 @@ end
 
 ## Nullable Arrays
 
-@compat abstract type AbstractNullableCategoricalArray{T, N, R} <: AbstractArray{Nullable{CategoricalValue{T, R}}, N} end
+@compat abstract type AbstractNullableCategoricalArray{T, N, R} <: AbstractArray{Union{CategoricalValue{T, R}, Null}, N} end
```

Comment:
So to remind myself, the idea you mentioned was to actually get rid of `AbstractNullableCategoricalArray`? And just have
```julia
@compat abstract type AbstractCategoricalArray{T, N, R} <: AbstractArray{Union{CategoricalValue{T, R}, Null}, N} end
```
?
I'm currently building out a `WeakRefStringArray` type and trying to get the subtyping right. `getindex` would then have a return type of `Union{WeakRefString{T}, Null}`.